upscale_swin2sr/upscale_images.py

```diff
@@ -0,0 +1,255 @@
+        embed_dim=180,
+        num_heads=[6, 6, 6, 6, 6, 6],
+        mlp_ratio=2,
+        upsampler="nearest+conv",
+        resi_connection="1conv",
+    )
+    pretrained_model = torch.load(MODEL_PATH)
+    model.load_state_dict(
+        pretrained_model[PARAM_KEY_G]
+        if PARAM_KEY_G in pretrained_model.keys()
+        else pretrained_model,
+        strict=True,
+    )
+    return model
+
+
+def filter_keys(key_set):
+    def _f(dictionary):
+        return {k: v for k, v in dictionary.items() if k in key_set}
+
+    return _f
+
+
+def get_dataloader(global_batch_size, num_workers, num_train_examples):
+    num_worker_batches = math.ceil(
+        num_train_examples / (global_batch_size * num_workers)
+    )  # per dataloader worker
+    resize = transforms.Resize((DOWNSAMPLE_TO, DOWNSAMPLE_TO))
+
+    def preprocess_images(sample):
+        # We need to ensure that the original and the edited images undergo the same
+        # augmentation transforms.
+        images = torch.stack(
+            [
+                transforms.ToTensor()(sample["original_image"]),
+                transforms.ToTensor()(sample["edited_image"]),
+            ]
+        )
+        transformed_images = resize(images)
+
+        # Separate the original and edited images.
+        original_image, edited_image = transformed_images.chunk(2)
+
+        # Pad the images.
+        def pad(img_lq):
+            _, _, h_old, w_old = img_lq.size()
+            h_pad = (h_old // WINDOW_SIZE + 1) * WINDOW_SIZE - h_old
+            w_pad = (w_old // WINDOW_SIZE + 1) * WINDOW_SIZE - w_old
+            img_lq = torch.cat([img_lq, torch.flip(img_lq, [2])], 2)[
+                :, :, : h_old + h_pad, :
+            ]
+            img_lq = torch.cat([img_lq, torch.flip(img_lq, [3])], 3)[
+                :, :, :, : w_old + w_pad
+            ]
+            return img_lq.squeeze(0)
+
+        original_image = pad(original_image)
+        edited_image = pad(edited_image)
+
+        return {
+            "original_image": original_image,
+            "edited_image": edited_image,
+            "edit_prompt": sample["edit_prompt"],
+            "original_prompt": sample["original_prompt"],
+        }
+
+    dataset = (
+        wds.WebDataset(DATASET_PATH, resampled=True, handler=wds.warn_and_continue)
+        .decode("pil", handler=wds.warn_and_continue)
+        .rename(
+            original_prompt="original_prompt.txt",
+            original_image="original_image.jpg",
+            edit_prompt="edit_prompt.txt",
+            edited_image="edited_image.jpg",
+            handler=wds.warn_and_continue,
+        )
+        .map(
+            filter_keys(
+                {"original_image", "edited_image", "edit_prompt", "original_prompt"}
+            ),
+            handler=wds.warn_and_continue,
+        )
+        .map(preprocess_images, handler=wds.warn_and_continue)
+        .batched(BATCH_SIZE, partial=False, collation_fn=default_collate)
```

Comment:
```suggestion
        .batched(BATCH_SIZE, partial=True, collation_fn=default_collate)
```

Since we want all batches.